maven-plugin/src/main/java/org/openmrs/maven/plugins/Setup.java

```diff
@@ -417,26 +371,112 @@ private void setDebugPort(Server server) {
         }
     }
 
-    private DistroProperties extractDistroToServer(Server server, boolean isCreatePlatform, File serverPath) throws MojoExecutionException, MojoFailureException {
-        DistroProperties distroProperties;
-        if(DistroHelper.isRefapp2_3_1orLower(server.getDistroArtifactId(), server.getVersion())){
-            distroProperties = new DistroProperties(server.getVersion());
-        } else {
-            distroProperties = distroHelper.downloadDistroProperties(serverPath, server);
+    private void setServerVersionsFromDistroProperties(Server server, DistroProperties distroProperties) throws MojoExecutionException {
+        if(server.getPlatformVersion() == null){
+            server.setPlatformVersion(distroProperties.getPlatformVersion(distroHelper, server.getServerDirectory()));
+        }
+        if(server.getVersion() == null){
+            server.setVersion(distroProperties.getVersion());
         }
-        moduleInstaller.installCoreModules(server, isCreatePlatform, distroProperties, distroHelper);
-        installOWAs(server, distroProperties);
-        return distroProperties;
     }
 
-    private void configureVersion(Server server, DistroProperties distroProperties) throws MojoExecutionException {
-        if (distroProperties != null) {
-            if(server.getPlatformVersion() == null){
-                server.setPlatformVersion(distroProperties.getPlatformVersion(distroHelper, server.getServerDirectory()));
+    private void setupDatabase(Server server, DistroProperties distroProperties) throws MojoExecutionException {
```

Comment:
`setupDatabase` and `setupDatabaseForServer` were pretty much just copy-pasted straight out of `setup`. No refactoring or changes.